### PR TITLE
Feature/hexxle 170 placement animations

### PR DESCRIPTION
**Pull request by denadex**

### CODE REVIEW

**Review by RicardoMonteiroSimoes — APPROVED** [review]
I was under the impression that my part had some "randomized" delay to each tile being placed, which made it look a bit more fluent. Was there any concrete reason for not using that?

**Comment by N2Drew** [comment]
Looks good to me. I actually like the more streamlined animations better.

**Review by N2Drew — APPROVED** [review]
Looks good to me. I actually like the more streamlined animations better.

**Comment by denadex** [comment]
@RicardoMonteiroSimoes Wasn't sure if I should add them, since the task only asked for animation but I can add them tomorrow, if you wish

**Comment by denadex** [comment]
@DennisBannerman pls wait with merging, I'm adding @RicardoMonteiroSimoes delay function to this branch

**Comment by denadex** [comment]
nvm xD

**Comment by DennisBannerman** [comment]
Yeah about that hahaha...

**Comment by denadex** [comment]
no worries

**Comment by DennisBannerman** [comment]
Only rebased it though, hopefully you can still manage to implement your stuff.

**Comment by denadex** [comment]
@DennisBannerman do you wanna have a look at it or should I just send it?

**Comment by DennisBannerman** [comment]
> 
> 
> @DennisBannerman do you wanna have a look at it or should I just send it?

eh, give me two minutes

File: unity/Assets/Scripts/Unity/UnityMap.cs [hunk 1]
```diff
@@ -104,6 +105,15 @@ public GameObject TileObjectOnPoint(Vector3 point)
...
         public void OnTilePlaced(object sender, TileMapEventArgs<ITile> e)
+        {
+            ITile tile = e.Tile;
+            GameObject template = TileTemplate;
+            Material material;
+            material = materials[(int)e.Tile.Type.Type - 1];
+            StartCoroutine(PlaceTileDelayed(tile, template, material, CoordinateToPoint(tile.Coordinate), Quaternion.Euler(0, 0, 0), e.Tile.Type.Type));
+        }
+
+ /*       public void OnTilePlaced(object sender, TileMapEventArgs<ITile> e)
```
Comment (DennisBannerman) [inline unity/Assets/Scripts/Unity/UnityMap.cs:116]: If it's not used, delete it. You don't have to worry about keeping "old" stuff just to be able to go back to it. We can always look it up via git.

File: unity/Assets/Scripts/Unity/UnityMap.cs [hunk 2]
```diff
@@ -157,5 +167,23 @@ public List<GameObject> GetAffectedTiles(Coordinate coordinate)
             }
             return new List<GameObject>();
         }
+
+        IEnumerator PlaceTileDelayed(ITile tile, GameObject gameObject, Material material, Vector3 coordinate, Quaternion quaternion, EType type)
```
Comment (DennisBannerman) [inline unity/Assets/Scripts/Unity/UnityMap.cs:171]: Can we make this private? What's the IEnumerator return type for?

**Review by DennisBannerman — APPROVED** [review]
Looks good! We'll probably have to look at the animation/camera interaction in the future, since the animation doesn't really show well, if you zoom out too much, but that's a problem for another time.

File: unity/Assets/Scripts/Unity/UnityMap.cs (hunk 2 above)
Comment (DennisBannerman) [reply unity/Assets/Scripts/Unity/UnityMap.cs:171]: Ok, you need an IEnumerator for the Coroutines, I see. All good then.

Comment (DennisBannerman) [reply unity/Assets/Scripts/Unity/UnityMap.cs:171]: Ok, last point: Don't think you actually need line 175. The tile sets it's own material anyway.

That means you can get rid of a bunch of parameters for the method, namely all but 'tile' and 'type'.

**Comment by denadex** [comment]
Does the great coding god @DennisBannerman approve my offer, now?

**Comment by DennisBannerman** [comment]
Thou shalt be granted a mortal merge!

**Pull request merged**